facp.md

```diff
@@ -0,0 +1,217 @@
+  addr_type_t X_GPE1_BLK;
+  addr_type_t SLEEP_CONTROL_REG;
+  addr_type_t SLEEP_STATUS_REG;
+  uint64_t hv_vendor_identity;
+  //More going to be added
+}
+```
+.
+It contains the header, the DSDT pointer, and blocks.
+The blocks will be used for shutdown.
+
+### Shutdown via FACP
+The ACPI shutdown is known as easy as several couple lines, however, enablement and initialization is neccessary for the shutdown code.
+
+I. Enabling the ACPI
+
+Enabling the ACPI is done with the following steps:
+
+1. Check if ACPI is enabled
+2. Check if ACPI is enabled
+3. If not-->enable ACPI, If yes-->stop because it is already
+4. Check if ACPI is enabled
+5. If yes-->Enable ACPI, If no-->send error
+
+The first check, step 1, could be done by comparing inw PM1a CNT Block &(AND operator) SCI_EN and 0
+
+Code: `inw((uint16_t) PM1a_CNT_BLK) & (uint16_t) SCI_EN) == 0`
+
+The next check can be done by checking if smi cmd and acpi enable is 0
+
+If yes, the ACPI is already enabled for you, if not, it isn't.
+
+Code again: `smi_cmd != 0 && acpi_enable != 0`
+
+Now it's time to enable the ACPI with a single line of code:
+
+`outb((uint16_t) smi_cmd, acpi_enable);`
+
+It outbytes data acpi enable to smi cmd port
+
+and finally we need to check if it enabled the ACPI. It can be done by 3 steps:
+
+1.
+```c
+int i;
+for(i = 0; i < 300; i++){
+	if(inw((uint16_t) PM1a_CNT_BLK & (uint16_t) SCI_EN) == 1){
+		break;
+	}
+		suspend(10);
+	}
+}
+```
+2. Do the same thing for the pm2b cnt block.
+3. Give error if not, give success message if is.
+
+II. ACPI initialization
+
+You must have found the FACP and have the header.
+
+S5 Address
+can be found by several lines of code basically doing a while
+
+But you need to first add 36 from the DSDT as the address, get the dsdt length as the following code, and finally do a while until DSDT Length -- is greater than 0
+While it does it, you should compare the first 4 characters via memcmp like functions and when you find it, you'll break and continue to the next step. Don't forget to add 1 every time to the address or you will not get the loop properly.
+```c
+	uint8_t* s5_addr = (uint8_t*) facp->dsdt + 36;
+	int dsdt_len = *(facp->dsdt + 1) - 36;
+	while(0 < dsdt_len--){
+		if(memcmp(s5_addr, "_S5_", 4)==0){
+			break;
+		}
+		s5_addr++;
+	}
+```
+Now that you have found
+
+the S5 addr
+
+then you will go intitialize everything.
+
+All these things will happen given that the DSDT length is greater than 0, which mean it has any data so you might want to get a while for this thing.
+
+First you need to check if AML is valid.
+
+
+![image](https://user-images.githubusercontent.com/39773400/156290876-ddae62f7-2f28-4f84-8a42-d28e5ebe370a.png)
+
+Figure 1
+
+![image](https://user-images.githubusercontent.com/39773400/156291769-5b5cf088-f902-431e-8c23-fa732fa12067.png)
+
+Figure 2
+
+Figure 1 and 2 explains the encoding value of the encoding name and how all these look like repectively as a photo.
+
+Your code must look like this:
+```c
```

Comment:
maybe it is explained abouve (i haven't looked at the picture) but again there is lot of code without an explanation. 

Just looking at it i have more questions than answers: 
* What is SLP_TYPb? 
* Why == 0x08 ot 0x12 ? why the \\ ? 
* Why 

The idea of these notes 9at least this is my opinion) is that people should be able to code stuff we explain without a line of code from our side (or very little) . Is ok keeping this code here but it has to be always completely explained, and who is reading should be able to write something similar without the need of looking at it or copying and pasting it.